### PR TITLE
Add form validation

### LINKED ISSUE #64: Improve Organization Creation Form
# Improve Organization Creation Form
Add more features to the organization creation form to improve the user experience and ensure data is input correctly.

## Why?
As it stands, the organization creation form is extremely bare bones and does not have many specific requirements, which means that there may not be consistency in organization generation.

## Acceptance Criteria
- [ ] Autogenerate slugs based on organization name (remove non-alphanumeric characters, abbreviate if possible, etc.)
- [ ] Require emails, social media links, and logos to be in the proper format.
- [ ] Require the short and long description to have a min and max character length.

### PR DESCRIPTION
**Pull request by atoneyd**
This PR adds more entry validation to the organization editor/creation form.

# Major Changes
- Added character limit of 150 to short description
- Added character limit of 2000 to long description
- Added valid email check
- Enforcing slugs to be limited to lowercase letters, numbers, and dashes
- Org Details page will default to short description if no long description is provided

# Testing
- Run `honcho start`
- Log in as Admin
- Navigate to organization editor
- Test new validations

# Developer notes
Ticket requested validation for links; not something we can really do with just regex since almost anything can be a link. Did not modify anything for now but I'm open to discussion if we still want to implement this.

Closes #64

### CODE REVIEW

**Review by briannata — CHANGES_REQUESTED** [review]
LGTM, but just a few nits!

Note: we should discuss whether we want to continue using links for images or change to an image upload

File: frontend/src/app/organization/organization-editor/organization-editor.component.html [hunk 1]
```diff
@@ -11,26 +11,26 @@
...
                 <mat-label>Organization Name</mat-label>
-                <input matInput placeholder="" formControlName="name" name="name" required>
+                <input matInput placeholder="" formControlName="name" name="name">
             </mat-form-field>
             <!-- Slug Field -->
-            <mat-form-field appearance="fill" color="accent">
+            <mat-form-field hintLabel="Must contain only letters, numbers, or dashes" appearance="fill" color="accent">
                 <mat-label>Slug</mat-label>
-                <input matInput placeholder="" formControlName="slug" name="slug" required>
+                <input matInput placeholder="" formControlName="slug" name="slug">
```
[the remark this answers is not in the document]
Comment (atoneyd) [reply frontend/src/app/organization/organization-editor/organization-editor.component.html:19]: I added validators in the TS file which already make it required

**Review by briannata — APPROVED** [review]
LGTM!

**Review by briannata — CHANGES_REQUESTED** [review]
Can you add it so that the organization details page uses the short description if the long description doesn't exist (since a long description isn't required in the creation form).

**Review by briannata — APPROVED** [review]
LGTM!

**Pull request merged**